lib/instrumentation/modules/fastify.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const semver = require('semver')
+
+module.exports = function (fastify, agent, version, enabled) {
+  if (!enabled) return fastify
+
+  agent.logger.debug('wrapping fastify build function')
+
+  return semver.gte(version, '2.0.0-rc') ? wrappedBuild2Plus : wrappedBuildPre2
+
+  function wrappedBuild2Plus () {
+    const _fastify = fastify.apply(null, arguments)
+
+    agent.logger.debug('adding onRequest hook to fastify')
+    _fastify.addHook('onRequest', (req, reply, next) => {
```

Comment:
I don't know how your internal implementation works, but if needed you can register also the `onResponse` hook, that will be fired once the Node.js's response fires the `finish` or the `error` events.
The signature of the hook in Fastify v2 is the same of the `onRequest` hook, so you can access the exact same data if needed :)

Reply:
I automatically get notified when the request starts and ends without a hook in Fastify because the agent hooks into the `http` module it self. What I need this hook for is knowing what to name the transaction - here I need access to the router. That's all I use this for.